cpp/src/arrow/flight/sql/column_metadata.cc

```diff
@@ -58,8 +58,10 @@ const char* ColumnMetadata::kIsSearchable = "ARROW:FLIGHT:SQL:IS_SEARCHABLE";
 const char* ColumnMetadata::kRemarks = "ARROW:FLIGHT:SQL:REMARKS";
 
 ColumnMetadata::ColumnMetadata(
-    std::shared_ptr<const arrow::KeyValueMetadata> metadata_map)
-    : metadata_map_(std::move(metadata_map)) {}
+    std::shared_ptr<const arrow::KeyValueMetadata> metadata_map) {
+  metadata_map_ = std::move(metadata_map ? metadata_map
```

Comment:
Seem like this line is the issue (https://github.com/apache/arrow/actions/runs/20586321192/job/59123338759).

Reply:
ack, I was questioning that line and should've checked the CI...let's revert for now

Reply:
https://github.com/apache/arrow/pull/48686